sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -125,39 +81,34 @@ object ScalaNativePluginInternal {
       "org.scala-native" % "nscplugin" % nativeVersion cross CrossVersion.full)
   )
 
-  lazy val globalSettings = Seq(
-    nativeWarnOldJVM := {
-      val logger = nativeLogger.value
-      Try(Class.forName("java.util.function.Function")).toOption match {
-        case None =>
-          logger.warn("Scala Native is only supported on Java 8 or newer.")
-        case Some(_) =>
-          ()
-      }
-    },
+  lazy val scalaNativeBaseSettings: Seq[Setting[_]] = Seq(
+    crossVersion := ScalaNativeCrossVersion.binary,
+    crossPlatform := NativePlatform,
     nativeClang := {
       val clang = discover("clang", clangVersions)
       checkThatClangIsRecentEnough(clang)
       clang
     },
+    nativeClang in Compile := nativeClang.value,
+    nativeClang in Test := nativeClang.value,
```

Comment:
It's true that those aliases are useless for `Test` and `Config` but we still need some of them for `NativeTest` to work.